### PR TITLE
Prototype that works with Olive generated PP output models

### PR DESCRIPTION
**Pull request by shaahji**
Prototype that works with Olive generated PP output models

### CODE REVIEW

File: examples/llama2_driver.py [hunk 1]
```diff
@@ -25,66 +20,83 @@ def _get_world_size() -> int:
...
+    config = ModelArgs.from_name(Path(args.checkpoint_path).parent.name)
+    config.n_local_layers = config.n_local_layers // world_size
 
-        torch.cuda.set_device(rank)
-        dist.init_process_group(backend="nccl", rank=rank, world_size=world_size)
-        model_path = f'models/model_{rank:02d}.pt'
-    else:
-        torch.cuda.set_device(args.device)
-        model_path = f'models/model.pt'
+    with torch.device('meta'):
```
Comment (devang-ml) [inline examples/llama2_driver.py]: What is 'meta' device?

File: examples/llama2_driver.py [hunk 2]
```diff
@@ -93,10 +105,25 @@ def _main():
...
         print(tokenizer.decode(generated_tokens.tolist()))
 
     return 0
 
 if __name__ == '__main__':
     sys.exit(_main())
+
+# To create checkpoints from transformer model:
+#   Use gpt-fast
+#   ./scripts/prepare.sh meta-llama/Llama-2-7b-chat-hf
```
Comment (devang-ml) [inline examples/llama2_driver.py]: I am not sure, why do we need to create checkpoint? One would get llama2 checkpoint from meta.

Comment (shaahji) [reply examples/llama2_driver.py]: This step is also converting the weights from hf format to torch.nn.module format.

File: examples/llama2_driver.py (hunk 1 above)
Comment (shaahji) [reply examples/llama2_driver.py]: It is something to do with being able to create arbitrarily large tensors. May not be required for this specific example though. Some docs also call them "Fake Tensors".

Comment (devang-ml) [reply examples/llama2_driver.py]: then let's remove it

File: examples/llama2_driver.py (hunk 2 above)
Comment (devang-ml) [reply examples/llama2_driver.py]: Could we eliminate gpt-fast dependency?

Comment (shaahji) [reply examples/llama2_driver.py]: It's an independent script so we can surely pull that script into orchard. The original source of that script is not even gptfast. It comes from another open source project to convert between different weight formats.

Comment (devang-ml) [reply examples/llama2_driver.py]: Let's do pull the script into orchard.

File: examples/llama2_driver.py (hunk 1 above)
Comment (shaahji) [reply examples/llama2_driver.py]: Done!

File: examples/llama2_driver.py (hunk 2 above)
Comment (shaahji) [reply examples/llama2_driver.py]: Done!

**Review by devang-ml — APPROVED** [review]

**Comment by shaahji** [comment]
I don't have write permissions in orchard so I can't merge this change.

**Pull request merged**